rasa/core/actions/forms.py

```diff
@@ -666,6 +685,9 @@ async def is_done(
         domain: "Domain",
         events_so_far: List[Event],
     ) -> bool:
+        if any(isinstance(event, ActionExecutionRejected) for event in events_so_far):
+            return False
+
```

Comment:
I don't thin it will cause policies to predict different action, though, because `ActionExecuted(form_name)` will be added to `tracker.events`, no?
Also, isn't returning `False` here will deactivate the form? That's not what we want to do

Reply:
return `False` here means `is_done = False` which will **not** deactivate the form.

> I don't thin it will cause policies to predict different action, though, because ActionExecuted(form_name) will be added to tracker.events, no?

Good point. This will work as long as it's added as last event in the tracker. I'll make the test more robust.

Reply:
Arg. It's actually a bit more complicated than that. @ArjaanBuijk What are you trying to do by rejection but still setting slots?

Reply:
fixed. rejections in conjunction with other events are now possible